docs/tutorials/apiIntegration/userSponsored.md

```diff
@@ -0,0 +1,304 @@
+---
+sidebar_label: "User sponsored"
+sidebar_position: 1
+---
+
+# User sponsored
+This tutorial represents the API integration flow where smart contract wallet has the native tokens to sponsor the transaction.
+
+## Pre-requisites: 
+
+- Biconomy bundler url (refer to the [docs](/dashboard#bundler-keys) to get the same)
+- Partial user operation, where
+    - sender is the smart account Address
+    - nonce can be calculated using the smart account contract methods.
+    - initCode will be 0x, if the account is already deployed or can be fetched using contract methods.
+    - paymasterAndData will be 0x, given the smart account has native tokens to pay for the gasFees.
+    - callData is the abi encoded form of transaction.
+    - It also requires putting a semi-valid/ dummy signature (e.g. a signature in the right length).
+
+```ts
+type UserOperation = {
+  sender: string;
+  nonce: BigNumberish;
+  initCode: BytesLike;
+  callData: BytesLike;
+  callGasLimit: BigNumberish;
+  verificationGasLimit: BigNumberish;
+  preVerificationGas: BigNumberish;
+  maxFeePerGas: BigNumberish;
+  maxPriorityFeePerGas: BigNumberish;
+  paymasterAndData: BytesLike;
+  signature: BytesLike;
+}
+
+let partialUserOp: Partial<UserOperation> = {
+  sender: '0x4dF23B78543F5c2F9CBCDF09956288B3e97bb9a4',
+  nonce: '0x08',
+  initCode: "0x",
+  paymasterAndData: "0x",
+  callData: "0x0000189a000000000000000000000000322af0da66d00be980c7aa006377fcaaeee3bdfd000000000000000000000000000000000000000000000000002386f26fc1000000000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000000",
+  signature: "0x00000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000001c5b32F37F5beA87BDD5374eB2aC54eA8e000000000000000000000000000000000000000000000000000000000000004181d4b4981670cb18f99f0b4a66446df1bf5b204d24cfcb659bf38ba27a4359b5711649ec2423c5e1247245eba2964679b6a1dbb85c992ae40b9b00c6935b02ff1b00000000000000000000000000000000000000000000000000000000000000",
+}
+```
+
+## 1. Calculate Gas estimations: 
+Based on the response, update the userOp gas limits and values.
+
+```ts
+async function getGasEstimations (partialUserOp: Partial<UserOperation>) : Promise<UserOperation> {
+    const url="https://bundler.biconomy.io/api/v2/80001/nJPK7B3ru.dd7f7861-190d-41bd-af80-6877f74b8f44"
+    
+    const data =
+    {
+      "method": "eth_estimateUserOperationGas",
+      "params": [
```

Comment:
This can be made cleaner by just doing this:
`"params": [partialUserOp, "0x5ff137d4b0fdcd49dca30c7cf57e578a026d2789"]`